src/Tests/Eshopworld.Messaging.Tests/MessengerQueueTest.cs

```diff
@@ -169,6 +171,52 @@ public async Task Test_LockAbandon_MessageFlow()
         }
     }
 
+    [Fact, IsLayer1]
+    public async Task Test_Concurrency_MessageFlow()
+    {
+        await ServiceBusFixture.ServiceBusNamespace.ScorchNamespace();
+
+        using (var ts = new CancellationTokenSource())
+        using (var msn = new Messenger(ServiceBusFixture.ConfigSettings.ServiceBusConnectionString, ServiceBusFixture.ConfigSettings.AzureSubscriptionId))
+        {
+
+            var sendTasks = new List<Task>();
+            for (var i = 0; i < 2; i++)

```

Comment:
Maybe increase the count bit more to have more chances for concurrent access? 5? 10?

Reply:
I feel I don't need this as the approach taken was that I wrote this test beforehand to reproduce the same crash/exceptions of the bug report, then I applied the fix and verified the test passed.